### PR TITLE
vendor gopkg.in/yaml.v1

### PR DESCRIPTION
**Pull request by lsm5**
Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

### CODE REVIEW

**Comment by lsm5** [comment]
@rhatdan ptal

**Comment by lsm5** [comment]
@cgwalters ptal

**Comment by rhatdan** [comment]
I guess you have to vendor these in.  Can't just use the ones installed on the host?

**Comment by lsm5** [comment]
we could use the ones installed on the host for Fedora, and maybe CentOS, but RHEL would need these bundled

**Comment by cgwalters** [comment]
s/installed on the host/available in the repos/

However, the problem is:

http://pkgs.fedoraproject.org/cgit/rpms/golang-gopkg-yaml.git/tree/golang-gopkg-yaml.spec#n9

Which basically means if I try to build this for CentOS, it turns into an entirely empty and useless package.  Why are we doing that?

**Comment by ingvagabund** [comment]
Since the plan was no devel rpm for RHEL. That has changed. I am updating spec files and enabling devel meantime.

**Comment by ingvagabund** [comment]
RHEL here means epel7

**Comment by ingvagabund** [comment]
Will be the epel7 branch enough? I can git merge master is needed.

http://koji.fedoraproject.org/koji/buildinfo?buildID=789866

**Comment by cgwalters** [comment]
CAHC tracks dist-git's f24 branch by default, although we can use any other branch.  I'd prefer tracking f24, but master is OK.

**Comment by ingvagabund** [comment]
Oh, wrong build. It is yaml, not check. @cgwalters ok, will do

**Comment by lsm5** [comment]
Fedora/CentOS aside, RHEL will still need this

**Comment by ingvagabund** [comment]
It should be vendored anyway so one can run CI tests with the development deps. One can switch to de-bundled deps anytime as long as gopkg.in/yaml.v1's commit is know in advance.

**Comment by ingvagabund** [comment]
Which brings me to question, where is the Godeps.json (or similar file)?

**Comment by ingvagabund** [comment]
@cgwalters http://pkgs.fedoraproject.org/cgit/rpms/golang-gopkg-yaml.git/tree/golang-gopkg-yaml.spec#n9 updated

**Comment by lsm5** [comment]
ha :D last time I tried to vendor godbus/dbus, godep wasn't working as expected on fedora, let me retry

**Comment by cgwalters** [comment]
@ingvagabund but now the two cases are identical...so why have them at all?

**Comment by ingvagabund** [comment]
@cgwalters bundled for RHEL, un-bundled for the rest

**Comment by cgwalters** [comment]
@ingvagabund What I mean is:

```
diff --git a/golang-gopkg-yaml.spec b/golang-gopkg-yaml.spec
index 33a35db..c6a79ac 100644
--- a/golang-gopkg-yaml.spec
+++ b/golang-gopkg-yaml.spec
@@ -1,17 +1,8 @@
-%if 0%{?fedora} || 0%{?rhel} == 6
 %global with_devel 1
 %global with_bundled 0
 %global with_debug 0
-# both tests failing
 %global with_check 0
 %global with_unit_test 1
-%else
-%global with_devel 1
-%global with_bundled 0
-%global with_debug 0
-%global with_check 0
-%global with_unit_test 1
-%endif
 
 %if 0%{?with_debug}
 %global _dwz_low_mem_die_limit 0
```

**Comment by lsm5** [comment]
I'm hoping atleast for:

On Thu, Aug 25, 2016 at 08:29:42AM -0700, Colin Walters wrote:
> @ingvagabund What I mean is:
> 
> ```
> diff --git a/golang-gopkg-yaml.spec b/golang-gopkg-yaml.spec
> index 33a35db..c6a79ac 100644
> --- a/golang-gopkg-yaml.spec
> +++ b/golang-gopkg-yaml.spec
> @@ -1,17 +1,8 @@
> -%if 0%{?fedora} || 0%{?rhel} == 6
%if 0%{?fedora} || 0%{?centos} || 0%{?rhel} == 6
>  %global with_devel 1
>  %global with_bundled 0
>  %global with_debug 0
> -# both tests failing
>  %global with_check 0
>  %global with_unit_test 1
> -%else
> -%global with_devel 1
> -%global with_bundled 0
%global with_bundled 1
> -%global with_debug 0
> -%global with_check 0
> -%global with_unit_test 1
> -%endif
>  
>  %if 0%{?with_debug}
>  %global _dwz_low_mem_die_limit 0
> ```
> 
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/projectatomic/oci-register-machine/pull/27#issuecomment-242429962

-- 
Lokesh
Freenode: lsm5
GPG: 0xC7C3A0DD

File: Makefile [hunk 1]
```diff
@@ -22,7 +22,7 @@ GOBUILD=go build -a -ldflags "${LDFLAGS:-} -B 0x$(shell head -c20 /dev/urandom|o
 # Example:
 #   make build
 oci-register-machine: oci-register-machine.go
-	GOPATH=$GOPATH:/usr/share/gocode $(GOBUILD) -o oci-register-machine
+	GOPATH=$(GOPATH):/usr/share/gocode $(GOBUILD) -o oci-register-machine
```
Comment (cgwalters) [inline Makefile]: Hmm, I think we normally want $$GOPATH.

**Comment by cgwalters** [comment]
@rhatdan and I discussed this in person and we agreed to vendor/bundle for now and avoid the pain of the split dependency.

File: Makefile (hunk 1 above)
Comment (lsm5) [inline Makefile]: yup, just changed it, thanks

**Comment by ingvagabund** [comment]
@cgwalters I see [1]. This was a fastest hack. I need to find some time and redesign it.

[1] https://github.com/gofed/gofed/issues/113

**Pull request merged**